src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
I would leave the `throws Exception` on `CacheLoader`.
There is definitely the possibility that something can go wrong when loading a value, like the examples @hanslovsky has given.
The caches wrap these as `ExecutionExceptions`, which is also modeled after how Guava does it.
There is `Cache.unchecked()` which wraps a cache as an `UncheckedCache`, wrapping checked exceptions as unchecked `RuntimeException`s. This is used by the cached images, because getting a `Cell` in the `CellImg` cannot deal with exceptions. I think this should be good enough. From the point of view of `CellImg` removing `throws Exception` would result in the same thing, but leaving it at least maintains the flexibility to do something more specific in other applications of caches.